include/gridtools/fn/sid_neighbor_table.hpp

```diff
@@ -0,0 +1,68 @@
+/*
+ * GridTools
+ *
+ * Copyright (c) 2014-2022, ETH Zurich
+ * All rights reserved.
+ *
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+#pragma once
+
+#include <cstddef>
+#include <type_traits>
+
+#include "../common/array.hpp"
+#include "../fn/unstructured.hpp"
+#include "../sid/concept.hpp"
+
+namespace gridtools::fn::sid_neighbor_table {
+    namespace sid_neighbor_table_impl_ {
+        template <class IndexDimension,
+            class NeighborDimension,
+            std::size_t MaxNumNeighbors,
+            class PtrHolder,
+            class Strides>
+        struct sid_neighbor_table {
+            PtrHolder origin;
+            Strides strides;
+
+            GT_FUNCTION friend auto neighbor_table_neighbors(
+                sid_neighbor_table<IndexDimension, NeighborDimension, MaxNumNeighbors, PtrHolder, Strides> const &table,
+                std::size_t index) {
+
+                auto ptr = table.origin();
+                using element_type = std::remove_reference_t<decltype(*ptr)>;
+
+                gridtools::array<element_type, MaxNumNeighbors> neighbors;
+
+                sid::shift(ptr, sid::get_stride<IndexDimension>(table.strides), index);
+                for (std::size_t element_idx = 0; element_idx < MaxNumNeighbors; ++element_idx) {
+                    neighbors[element_idx] = *ptr;
+                    sid::shift(ptr, sid::get_stride<NeighborDimension>(table.strides), 1);
```

Comment:
```suggestion
                    sid::shift(ptr, sid::get_stride<NeighborDimension>(table.strides), 1_c);
```
Probably there is no performance difference as every reasonable compiler should be able to apply proper constant propagation here, but by using an `integral_constant`, we can enforce it (requires `using namespace gridtools::literals`).

Reply:
Hmm, I'd prefer not to mix in metaprogramming where not needed.

Reply:
I vote for applying the change, as it is not less readable and clearly expresses that this should be a compile time offset..

Reply:
I agree with @havogt. First, I don’t see why anybody could think that `_c` defines a complex number, standard literals are `i` etc. And second, yes, to write or understand such code, the user has to understand what `1_c` is and has to understand how SIDs work and what arguments must be passed to each function. But it’s much easier to understand that you should add a `_c` whenever possible than understanding when omitting it actually totally kills performance (due to failing auto-vectorization, aliasing-analysis, scalar replacement etc.).